src/dialects/mssql/query-interface.js

```diff
@@ -95,4 +95,47 @@ export class MsSqlQueryInterface extends AbstractQueryInterface {
 
     return await this.sequelize.queryRaw(sql, options);
   }
+
+  /**
+   * @override
+   */
+  async getForeignKeysForTables(tableNames, options) {
+    if (tableNames.length === 0) {
+      return {};
+    }
+
+    options = { ...options, type: QueryTypes.FOREIGNKEYS };
+
+    const results = await Promise.all(tableNames.map(tableName => this.sequelize.queryRaw(this.queryGenerator.getForeignKeysQuery(tableName, this.sequelize.config.database), options)));
```

Comment:
These two overrides are done so the database is passed on, which is not needed for the other dialects